File: java/frozenPizza/src/test/java/FrozenPizzaServiceShould.java
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertThrows;

public class FrozenPizzaServiceShould {

    @Test
    public void takeOrder() {
        Map<String, Integer> pizzaStock = new HashMap<>();
        pizzaStock.put("Margarita", 10);
        var sut = new FrozenPizzaService(pizzaStock);
        var order = sut.takeOrder("Margarita", 1);
        assert ("Margarita".equals(order.name()));
        assert (1 == order.quantity());
    }

    @Nested
    class Fail {
        private FrozenPizzaService sut = null;

        @BeforeEach
        public void setUp() {

            Map<String, Integer> pizzaStock = new HashMap<>();
            pizzaStock.put("Margarita", 10);
            this.sut = new FrozenPizzaService(pizzaStock);
        }

        @Test
        public void toOrderUnknowPizza() {
            assertThrows(PizzaNotFoundException.class, () -> sut.takeOrder("Pepperoni", 1));
        }

        @Test
        public void toOrderPizzaWithInsufficientStock() {
            assertThrows(PizzaOutOfStockException.class, () -> sut.takeOrder("Margarita", 20));
        }
    }
}

File: csharp/FrozenPizza.Tests/FrozenPizzaServiceShould.cs
using FluentAssertions;

namespace FrozenPizza.Tests;

public class FrozenPizzaServiceShould
{
    [Fact]
    public void TakeOrder()
    {
        Dictionary<string, int> stock = new()
        {
            { "Margarita", 10 }
        };

        var sut = new FrozenPizzaService(stock);
        var order = sut.TakeOrder("Margarita", 1);
        order.Should().NotBeNull()
            .And.BeEquivalentTo(new Order("Margarita", 1));
    }

    public class Fail
    {
        private readonly FrozenPizzaService _sut;

        public Fail()
        {
            Dictionary<string, int> stock = new()
            {
                { "Margarita", 10 }
            };

            _sut = new FrozenPizzaService(stock);
        }

        [Fact]
        public void ToOrderUnknowPizza()
        {
            _sut.Invoking(s => s.TakeOrder("Pepperoni", 1))
                .Should().Throw<PizzaNotFoundException>()
                .WithMessage("Pizza not found");
        }
        
        [Fact]
        public void ToOrderPizzaWithInsufficientStock()
        {
            _sut.Invoking(s => s.TakeOrder("Margarita", 20))
                .Should().Throw<PizzaOutOfStockException>()
                .WithMessage("Pizza out of stock");
        }
    }
}